### PR TITLE
Switching to minimal TFMs

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl

**Comment by davidfowl** [comment]
:shipit: I need to investigate why you needed to add System.Runtime.

**Comment by khellang** [comment]
:heart:

File: src/Microsoft.Extensions.DependencyInjection.Abstractions/ServiceDescriptor.cs [hunk 1]
```diff
@@ -123,7 +123,7 @@ internal Type GetImplementationType()
                 return typeArguments[1];
             }
 
-            Debug.Fail("ImplementationType, ImplementationInstance or ImplementationFactory must be non null");
+            Debug.Assert(false, "ImplementationType, ImplementationInstance or ImplementationFactory must be non null");
```
Comment (halter73) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/ServiceDescriptor.cs]: What platform doesn't have Debug.Fail btw?

Comment (pranavkm) [inline src/Microsoft.Extensions.DependencyInjection.Abstractions/ServiceDescriptor.cs]: netstandard1.0 which is the only TFM we're targeting in this package.

**Pull request merged**